### PR TITLE
Native Mobile UI Best Practices

### PR DESCRIPTION
**Pull request by ConnorLand**
Danny and I will resolve last comments in this PR.

### CODE REVIEW

File: content/howto/front-end/ui-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+In apps where custom UX or UI is more important (for example B2C and B2B apps) the following combinations are often employed:
+
+* **UI design + UX design** — This configuration is often used if an in-house UX team is used or a design agency is hired.
+* **UI design + UX design + Front-end development** — UX or UI experts that also have the skills to implement their designs and UX are key for this configuration. For example, several people in Mendix Expert Services are of these type.
+* **Mendix development + Front-end development** — A Mendix developer who knows how to implement styling, but does not have the “feeling” to create something beautiful. [todo: ask for clarity on this one. I don't understand how it's a combo]()
```
Comment (ConnorLand) [inline content/howto/front-end/ui-best-practices.md:37]: Could you clarify this bullet a bit? I'm a little confused by the word "feeling". Maybe something like this? "A Mendix developer who knows how to implement styling, but wants artistic help from a front-end development specialist."

File: content/howto/front-end/ui-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+## 3 Process
+
+At Mendix we strongly believe in enabling other developers as they grow to scale and speed up development. In general there are more Mendix developers than UI or UX designers. Therefore, when building a mobile app it is important that a UI or UX designer can enable Mendix developers as much as possible to speed up development.
+
+One could compare building Mendix to building lego. Give a maker a set of generic blocks that can be combined, and they can build anything. If they needed to constantly go to the store to get more blocks, their building would be slow and hampered. [todo: is it key we mention the Lego brand here? if not switch to "building blocks". I'm uncertain about this whole paragraph]()
```
Comment (ConnorLand) [inline content/howto/front-end/ui-best-practices.md:43]: I'd like to leave the Lego brand out of the docs. Also the "go to the store" bit confused me a little, so I tried to tweak the paragraph a bit:

"Building apps with Mendix is like building with toy building blocks. Give a developer a set of generic blocks that can be combined, and they can build anything. It is important to keep the "blocks" as generic and flexible as possible. You will see further examples of this thinking in the [Configure Design Properties](##configure-design-props) section below."

File: content/howto/front-end/ui-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+To learn about building a Mendix native mobile app's UI, consult the sections below. Be sure to see the [Configure Design Properties](#configure-design-props) section below for key information on implementing design easily.
+
+### 3.1. Create App Designs Using a UI Kit
+
+By designing a set of re-usable components including variations [todo:variations?](), a UI Designer can create a UI kit that can be used to design pages. This includes settings like the brand colors, typography, spacing settings, and the UI components (for example buttons and inputs) including variations. Designers frequently use the [Sketch](https://www.sketch.com/) tool to complete these tasks with satisfying results. [todo: is the UI Kit an upper-case Mendix product?]()
```
Comment (ConnorLand) [inline content/howto/front-end/ui-best-practices.md:49]: What exactly is the UI Kit? If it's a Mendix product, I will put it as "Mendix UI Kit" throughout this doc.

File: content/howto/front-end/ui-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+The default Atlas theme comes pre-bundled with a set of layouts. If these do not fit your app’s design, you can create or customize the layouts and name them accordingly.
+
+#### 3.2.4 Create Building Blocks for Common UI Structures
+
+Apps often have reoccurring patterns on pages or more complex UI patterns. For these cases it is useful to create building blocks so that Mendix developers can easily apply reoccuring patterns. In building blocks, you can use custom class names in order to be more specific. For that properties that developers should be able to change in building blocks, use design properties as much as possible [todo: clarify this last sentence]().
+
+#### 3.2.5 Create Page Templates for Common Page Structures**
+
+Similar to buildings blocks, you can create standard page templates so that other [todo: ask danny: is this formulation best for showing the parent dev children devs relationship?]() developers can easily implement a type of page.
```
Comment (ConnorLand) [inline content/howto/front-end/ui-best-practices.md:81]: Is this paragraph written well to sort of show the "parent/children" relationship of the main "parent" Dev and the multiple "children" Mendix devs beneath them who make their own iterations of the "parent" app? I'd like to include a short sentence in the Roles section laying this arrangement out.

File: content/howto/front-end/ui-best-practices.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+### 4.2 Plan to Reuse as Much as Possible
+
+A common pitfall is to start directly implementing directly from designs without thinking about reuse. While implementing directly can make sense for small apps, this practice risks creating several custom classes, spawning more variations of components which can slow down development, and can create significant technical debt.
+
+Try to avoid very specific class names. Very specific class names are often only used once, although the styles are probably used at least a few times. Combine and rewrite classes where possible. This will keep the theme clean and simple, and makes it easier to create design properties and enable Mendix developers.
+
+### 4.3 Never Change Something in the Core Folder
+
+By changing styling in your app's **core** folder [todo: folder, correct?]() you risk incurring errors the next time you update update Atlas. Updating Atlas might also be more difficult.
```
Comment (ConnorLand) [inline content/howto/front-end/ui-best-practices.md:109]: "app's **core** folder" correct?

File: content/howto/front-end/ui-best-practices.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+A large part of your app's overall styling can be changed in `theme/styles/native/app/custom-variables.js`. However, if more changes are needed you could extend a widget’s default styling or define the default styling yourself. For more information, see [How to Extend Design Properties to Customize Your Studio Experience](/howto/front-end/extend-design-properties-to-customize).
+
+### 4.4 Favor Design Properties Over Including Properties
+
+Avoid including properties which can be set with design properties. By being careful with your properties, your apps will be easier for other developers to understand and change. For example, if you make a certain element always have a margin there is a change that this will not fit in certain designs. However, a the developer could easily set a margin using design properties. One particularly useful approach is providing a default spacing this will cover 90% of cases, and give the developer the option to overwrite it using design properties. [todo: note to self, check this section against Paper. I made a lot of changes.]()
```
Comment (ConnorLand) [inline content/howto/front-end/ui-best-practices.md:115]: Please check this paragraph Danny. It received heavy edits, and I want to make sure it carries the points you intended.

File: content/howto/mobile/ui-best-practices.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+In apps where custom UX or UI is more important (for example B2C and B2B apps) the following combinations are often employed:
+
+* **UI design + UX design** — This configuration is often used if an in-house UX team is used or a design agency is hired.
+* **UI design + UX design + Front-end development** — UX or UI experts that also have the skills to implement their designs and UX are key for this configuration. For example, several people in Mendix Expert Services are of these type.
+* **Mendix development + Front-end development** — A Mendix developer who knows how to implement styling, but does not have the “feeling” to create something beautiful. [todo: ask for clarity on this one. I don't understand how it's a combo]()
```
Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:37]: Could you clarify this bullet a bit? I'm a little confused by the word "feeling". Maybe something like this? "A Mendix developer who knows how to implement styling, but wants artistic help from a front-end development specialist."

File: content/howto/mobile/ui-best-practices.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+## 3 Process
+
+At Mendix we strongly believe in enabling other developers as they grow to scale and speed up development. In general there are more Mendix developers than UI or UX designers. Therefore, when building a mobile app it is important that a UI or UX designer can enable Mendix developers as much as possible to speed up development.
+
+One could compare building Mendix to building lego. Give a maker a set of generic blocks that can be combined, and they can build anything. If they needed to constantly go to the store to get more blocks, their building would be slow and hampered. [todo: is it key we mention the Lego brand here? if not switch to "building blocks". I'm uncertain about this whole paragraph]()
```
Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:43]: I'd like to leave the Lego brand out of the docs. Also the "go to the store" bit confused me a little, so I tried to tweak the paragraph a bit:

"Building apps with Mendix is like building with toy building blocks. Give a developer a set of generic blocks that can be combined, and they can build anything. It is important to keep the "blocks" as generic and flexible as possible. You will see further examples of this thinking in the Configure Design Properties section below."

File: content/howto/mobile/ui-best-practices.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+To learn about building a Mendix native mobile app's UI, consult the sections below. Be sure to see the [Configure Design Properties](#configure-design-props) section below for key information on implementing design easily.
+
+### 3.1. Create App Designs Using a UI Kit
+
+By designing a set of re-usable components including variations [todo:variations?](), a UI Designer can create a UI kit that can be used to design pages. This includes settings like the brand colors, typography, spacing settings, and the UI components (for example buttons and inputs) including variations. Designers frequently use the [Sketch](https://www.sketch.com/) tool to complete these tasks with satisfying results. [todo: is the UI Kit an upper-case Mendix product?]()
```
Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:49]: "including variations" correct?

Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:49]: What exactly is the UI Kit? If it's a Mendix product, I will put it as "Mendix UI Kit" throughout this doc.

File: content/howto/mobile/ui-best-practices.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
+The default Atlas theme comes pre-bundled with a set of layouts. If these do not fit your app’s design, you can create or customize the layouts and name them accordingly.
+
+#### 3.2.4 Create Building Blocks for Common UI Structures
+
+Apps often have reoccurring patterns on pages or more complex UI patterns. For these cases it is useful to create building blocks so that Mendix developers can easily apply reoccuring patterns. In building blocks, you can use custom class names in order to be more specific. For that properties that developers should be able to change in building blocks, use design properties as much as possible [todo: clarify this last sentence]().
+
+#### 3.2.5 Create Page Templates for Common Page Structures**
+
+Similar to buildings blocks, you can create standard page templates so that other [todo: ask danny: is this formulation best for showing the parent dev children devs relationship?]() developers can easily implement a type of page.
```
Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:81]: Is this paragraph written well to sort of show the "parent/children" relationship of the main "parent" Dev and the multiple "children" Mendix devs beneath them who make their own iterations of the "parent" app? I'd like to include a short sentence in the Roles section laying this arrangement out.

File: content/howto/mobile/ui-best-practices.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+
+### 4.2 Plan to Reuse as Much as Possible
+
+A common pitfall is to start directly implementing directly from designs without thinking about reuse. While implementing directly can make sense for small apps, this practice risks creating several custom classes, spawning more variations of components which can slow down development, and can create significant technical debt.
+
+Try to avoid very specific class names. Very specific class names are often only used once, although the styles are probably used at least a few times. Combine and rewrite classes where possible. This will keep the theme clean and simple, and makes it easier to create design properties and enable Mendix developers.
+
+### 4.3 Never Change Something in the Core Folder
+
+By changing styling in your app's **core** folder [todo: folder, correct?]() you risk incurring errors the next time you update update Atlas. Updating Atlas might also be more difficult.
```
Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:109]: "app's core folder" correct?

File: content/howto/mobile/ui-best-practices.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+A large part of your app's overall styling can be changed in `theme/styles/native/app/custom-variables.js`. However, if more changes are needed you could extend a widget’s default styling or define the default styling yourself. For more information, see [How to Extend Design Properties to Customize Your Studio Experience](/howto/front-end/extend-design-properties-to-customize).
+
+### 4.4 Favor Design Properties Over Including Properties
+
+Avoid including properties which can be set with design properties. By being careful with your properties, your apps will be easier for other developers to understand and change. For example, if you make a certain element always have a margin there is a change that this will not fit in certain designs. However, a the developer could easily set a margin using design properties. One particularly useful approach is providing a default spacing this will cover 90% of cases, and give the developer the option to overwrite it using design properties. [todo: note to self, check this section against Paper. I made a lot of changes.]()
```
Comment (ConnorLand) [inline content/howto/mobile/ui-best-practices.md:115]: Please check this paragraph Danny. It received heavy edits, and I want to make sure it carries the points you intended.

File: content/howto/front-end/ui-best-practices.md (hunk 1 above)
Comment (dannyroestmx) [reply content/howto/front-end/ui-best-practices.md]: Already changed it based on feedback

File: content/howto/front-end/ui-best-practices.md (hunk 2 above)
Comment (dannyroestmx) [reply content/howto/front-end/ui-best-practices.md]: rewrote it already

File: content/howto/front-end/ui-best-practices.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dannyroestmx) [reply content/howto/front-end/ui-best-practices.md]: yes

File: content/howto/front-end/ui-best-practices.md (hunk 4 above)
Comment (dannyroestmx) [reply content/howto/front-end/ui-best-practices.md]: rewrote a bit

File: content/howto/front-end/ui-best-practices.md (hunk 5 above)
Comment (dannyroestmx) [reply content/howto/front-end/ui-best-practices.md]: I rewrote it a bit

**Pull request merged**